docs/Developer Documentation/Security Manager Design/index.md

```diff
@@ -0,0 +1,74 @@
+# Security Manager Guide
+
+### What is the Security Manager?
+
+Secure communication in SDL Core is provided by the Security Manager interface, implemented in Core as the [SecurityManagerImpl](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/security_manager/src/security_manager_impl.cc).
+
+Under the hood the Security Manager uses OpenSSL to to complete handshakes and encrypt/decrypt data. All OpenSSL operations are abstracted away by the [Crypto Manager](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/security_manager/src/crypto_manager_impl.cc) which provides a factory for SSL Context objects, the ability to update certificates from Certificate Authorities and SSL error reporting. The SSL Context objects created by the Crypto Manager can be used to establish SSL connections, as well as to encrypt or decrypt data. Events within the Security Manager will be propagated to any Security Manager Listeners that register with the Security Manager. The Protocol Handler implementation in Core implements the SecurityManagerListener interface in order to send protocol layer responses to handshakes.
+
+### What events are generated?
+
+#### OnHandshakeDone
+
+When a handshake has been completed, the `OnHandshakeDone` is called on each registered `SecurityManagerListener`. This function is called with two parameters: the connection key, and a `HandshakeResult` enum value (one of Fail, Success, or CertExpired)
+
+### OnCertificateUpdateRequired
+
+When a handshake is initiated and the connection does not have a valid certificate `OnCertificateUpdateRequired` will be called on each registered `SecurityManagerListener`. This could be either because no certificate was supplied for the connection or because the certificate is expired.
```

Comment:
```suggestion
When a handshake is initiated and the connection does not have a valid certificate, `OnCertificateUpdateRequired` will be called on each registered [SecurityManagerListener](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/include/security_manager/security_manager_listener.h). This could be either because no certificate was supplied for the connection or because the certificate is expired.
```